generators/heroku/generator.mjs

```diff
@@ -126,11 +111,9 @@ export default class HerokuGenerator extends BaseGenerator {
 
   get prompting() {
     return {
-      askForApp: runAsync(function () {
-        const done = this.async();
-
+      async askForApp(done) {
         if (this.herokuAppName) {
-          ChildProcess.exec(`heroku apps:info --json ${this.herokuAppName}`, (err, stdout) => {
+          this.spawnCommand(`heroku apps:info --json ${this.herokuAppName}`, (err, stdout) => {
             if (err) {
               this.abort = true;
```

Comment:
`this.abort` seems to be designed to don't execute the generator anymore.
This can be done by calling:
```suggestion
              this.cancelCancellableTasks();
```

But this looks an error why not throw an error?